### DISCUSSION TITLE
Is it needed to force user refresh token when you use firestore?

### DISCUSSION THREAD

**Discussion by jesus-mg-ios** [firebase/firebase-ios-sdk#11178] [Q&A] [answered]
Hi:

Right now I'm using firestore SDK (no the api REST) in a long life iOS software. I'm wondering if it's needed to refresh the user token or firestore piece does it by itself. 

Thanks in advance

**Comment by morganchen12** [comment, marked as answer]
The user token only needs to be refreshed for account-sensitive operations. You can, if you'd like, manually determine if your Firestore directory contains user-sensitive data and then require a user login refresh before accessing or modifying that data.
